src/components/CommentContainer.tsx

```diff
@@ -0,0 +1,66 @@
+import { useRouter } from 'next/router';
+import React, { useEffect, useRef, useState } from 'react';
+import { getComments, postComment } from '../api/api';
+import { commentType } from '../api/apiType';
+import Comment from './Comment';
+import CommentInput from './CommentInput';
+import NoComment from './NoComment';
+
+const CommentContainer = () => {
+  const router = useRouter();
+  const [comments, setComments] = useState<commentType[] | []>([]);
+  const id = router.query['id'];
+  const ref = useRef<HTMLInputElement>(null);
+
+  const saveComment = async () => {
+    try {
+      const result = await getComments(id as string, 5);
+      setComments(result);
+    } catch (error) {}
+  };
+
+  const onChangeHandler = (e: React.ChangeEvent<HTMLInputElement>) => {
+    if (ref.current) {
+      ref.current.value = e.target.value;
+    }
+  };
+
+  const registerCommentHandler = async () => {
+    const content = ref.current?.value;
+    try {
+      await postComment(id as string, content as string);
+    } catch (error) {
+    } finally {
+      window.location.reload();
+    }
+  };
+
+  useEffect(() => {
+    saveComment();
+  }, []);
```

Comment:
```suggestion
  useEffect(() => {
    if(id){
    saveComment();
    }
  }, []);
```
위처럼 바꾸시면 함수에 타입단언 쓰지 않아도 될 것 같아요

Reply:
넵